app/views/portal/show.rb

```diff
@@ -412,12 +417,118 @@ def content
               }
             end
           },
+          named_entities: named_entity_data,
           hierarchy: @hierarchy.blank? ? nil : record_hierarchy(@hierarchy),
           thumbnail: render_document_show_field_value(document, 'europeanaAggregation.edmPreview', tag: false)
         }.reverse_merge(helpers.content)
       end
     end
 
+    def named_entity_data()
+      data = [collect_concept_labels, collect_agent_labels, collect_time_labels, collect_place_labels]
+      present = false
+      for group in data
+        if group[:fields].size > 0
+          present = true
+        end
+      end
+      {
+        present: present,
+        data: data
+      }
+    end
+
+    def collect_agent_labels()
+      fields = []
+      agents = document.fetch('concepts', [])
+      (agents).map do |agent|
+        fields << { key: t('site.object.named-entities.who.term'), val: agent[:about], agt_link: true}
+        fields << { key: t('site.object.named-entities.who.label'), vals: normalise_named_entity(agent[:prefLabel]), multi: true }
+      end
+      {
+        title: t('site.object.named-entities.who.title'),
+        fields: fields,
+        present: fields.size > 0
+      }
+    end
+
+    def collect_place_labels()
+      fields = []
+      places = document.fetch('places', [])
+      (places).map do |place|
+        fields << { key: t('site.object.named-entities.where.term'), val: place[:about], agt_link: true}
+        fields << { key: t('site.object.named-entities.where.label'), vals: normalise_named_entity(place[:prefLabel]), multi: true }
+        if !place[:latitude].nil?
+          fields << { key: t('site.object.named-entities.where.latitude'), val: place[:latitude] }
+        end
+        if !place[:longitude].nil?
+          fields << { key: t('site.object.named-entities.where.longitude'), val: place[:longitude] }
+        end
+      end
+      {
+        title: t('site.object.named-entities.where.title'),
+        fields: fields,
+        present: fields.size > 0
+      }
+    end
+
+    def collect_time_labels()
```

Comment:
Assignment Branch Condition size for collect_time_labels is too high. [28.14/15]<br>Method has too many lines. [17/10]<br>Omit the parentheses in defs when the method doesn't accept any arguments.